pkgs/applications/misc/hello/default.nix

```diff
@@ -1,11 +1,11 @@
-{ lib, stdenv, fetchurl }:
+{ lib, runCommand, stdenv, fetchurl }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation ({ this, ... }: {
```

Comment:
i don't like the overhead this adds. the derivation gets a lot less readable

Reply:
I've reintroduced the `rec` keyword, so the changes are minimal. It wasn't unreadable before and now it's barely changed at all.

Now `version` is in scope again, which, if you override it won't affect `src`, as usual. That's the way derivations are currently written. In my opinion, `this.version` has better behavior, you don't have to agree and I'm ok to stick to the old format until the community decides otherwise. I'm only interested in making `passthru` behave correctly. The ability to use `src = f this.version;` is just side-catch.